test/test-define-x-expander-use-site-scope.rkt

```diff
@@ -0,0 +1,73 @@
+#lang racket
+
+(require syntax/parse
+         syntax/parse/experimental/eh
+         generic-syntax-expanders
+         syntax/stx
+         (for-syntax syntax/parse
+                     racket/syntax)
+         rackunit)
+
+(define-expander-type eh-mixin)
+
+(begin-for-syntax
+  (define eh-post-accumulate (make-parameter #f)))
+
+(define-for-syntax (strip-use-site stx)
+  (define bd
+    (syntax-local-identifier-as-binding (syntax-local-introduce #'here)))
+  (define delta
+    (make-syntax-delta-introducer (syntax-local-introduce #'here) bd))
+  (delta stx 'remove))
+
+(define-syntax define-eh-alternative-mixin
+  (syntax-parser
+    [(_ name ((~literal pattern) pat) ... (~optional (~seq #:post post)))
+     (let ()
+       #`(define-eh-mixin-expander name
+           (λ (_)
+             #,@(if (attribute post)
+                    #`(((eh-post-accumulate) (quote-syntax post)))
```

Comment:
No, that would insert #<void> in the resulting syntax. Furthermore, `eh-post-accumulate` is a parameter containing the dynamically-scoped accumulator function.

Calling it directly with `(eh-post-accumulate (quote-syntax post))` would set the parameter to a piece of syntax, instead of calling the function it contains.

For the use of `(quote-syntax post)` instead of `#'post`, it's slightly faster and safer — I'm injecting the syntax contained by the pattern variable `post` in the generated expander `name`, and quoting it there. A common issue if I were to use `#'post` is when `post` happens to contain ellipses specified by the user, these will cause havoc. For example, if the user wrote

```
#:post (~fail #:when (and (attribute some-pat) (not (stx-null? #'(other-pat ...)))))
```

then the resulting code will be
```
((eh-post-accumulate) #'(~fail #:when (and (attribute some-pat) (not (stx-null? #'(other-pat ...))))))
```
However, at that point, `other-pat` is not yet a pattern variable with ellipses, as we're simply quoting it to inject it later. The `#'` would nevertheless see a piece of syntax of the form `#'(blah #'(other-pat ...))` and will complain about ` no pattern variables before ellipsis in template`.

On the other hand, `quote-syntax` does not do any fancy treatment of ellipses or pattern variables, so it's the right tool for the job here — in the expanded code, we need to quote the syntax which is already there, verbatim. One could use `#'(... post)` to escape the ellipses appearing in the expanded form of `post`, but its safer (and faster) to use `quote-syntax` directly.

Macro-defining macros are always so tricky :)